pipeline/src/concluder/mod.rs

```diff
@@ -0,0 +1,307 @@
+pub use crate::since::SubscribeSinceMsg;
+pub use event::{
+    conclusion_events_to_record_batch, ConclusionData, ConclusionEvent, ConclusionInit,
+    ConclusionTime,
+};
+pub use table::ConclusionFeed;
+
+const CONCLUSION_EVENTS_TABLE: &str = "ceramic.v0.conclusion_events";
+const CONCLUSION_EVENTS_STREAM_TABLE: &str = "ceramic.v0.conclusion_events_stream";
+
+/// Concluder is responsible for making conclusions about raw events and publishing
+/// conclusion_events.
+#[derive(Actor)]
+pub struct Concluder {
+    ctx: SessionContextRef,
+    broadcast_tx: broadcast::Sender<RecordBatch>,
+}
+impl Concluder {
+    /// Create and spawn a new Concluder
+    pub async fn spawn_new<F: ConclusionFeed + 'static>(
+        size: usize,
+        ctx: &PipelineContext,
+        feed: ConclusionFeedSource<F>,
+        shutdown: Shutdown,
+    ) -> Result<(ConcluderHandle, Vec<JoinHandle<()>>)> {
+        let (broadcast_tx, _broadcast_rx) = broadcast::channel(1_000);
+        let actor = Concluder {
+            ctx: ctx.session(),
+            broadcast_tx,
+        };
+        let (handle, task_handle) = Self::spawn(size, actor, shutdown.wait_fut());
+        // Register tables
+        let last_processed_index = match feed {
+            ConclusionFeedSource::Direct(conclusion_feed) => {
+                ctx.session().register_table(
+                    CONCLUSION_EVENTS_TABLE,
+                    Arc::new(FeedTable::new(conclusion_feed.clone())),
+                )?;
+                conclusion_feed.max_highwater_mark().await?
+            }
+            #[cfg(test)]
+            ConclusionFeedSource::InMemory(table) => {
+                assert_eq!(
+                    crate::schemas::conclusion_events(),
+                    datafusion::catalog::TableProvider::schema(&table)
+                );
+                ctx.session()
+                    .register_table(CONCLUSION_EVENTS_TABLE, Arc::new(table))?;
+                let batches = ctx
+                    .session()
+                    .table(CONCLUSION_EVENTS_TABLE)
+                    .await?
+                    .aggregate(
+                        vec![],
+                        vec![datafusion::functions_aggregate::min_max::max(col("index"))
+                            .alias("max_index")],
+                    )?
+                    .collect()
+                    .await?;
+                batches.get(0).and_then(|batch| {
+                    batch.column_by_name("max_index").and_then(|index_col| {
+                        as_uint64_array(&index_col).ok().and_then(|index_col| {
+                            arrow::array::Array::is_valid(&index_col, 0).then(|| index_col.value(0))
+                        })
+                    })
+                })
+            }
+        };
+        ctx.session()
+            .register_table(
+                CONCLUSION_EVENTS_STREAM_TABLE,
+                Arc::new(StreamTable::new(handle.clone())),
```

Comment:
This is how consumers can get a stream of data. The `StreamTable` implements the datafusion traits to be a table, then it calls into the actors (via the handle) to subscribe to that actors data. 

There are a few other touch points I'll call out below.